src/App/Pages/Accounts/LoginPasswordlessRequestViewModel.cs

```diff
@@ -254,31 +265,66 @@ private async Task CheckLoginRequestStatus()
+        }
+
         private async Task CreatePasswordlessLoginAsync()
         {
             await Device.InvokeOnMainThreadAsync(() => _deviceActionService.ShowLoadingAsync(AppResources.Loading));
 
-            var response = await _authService.PasswordlessCreateLoginRequestAsync(_email, AuthRequestType);
-            if (response != null)
+            PasswordlessLoginResponse response = null;
+            var pendingRequest = await _stateService.GetPendingAdminAuthRequestAsync();
+            if (pendingRequest != null && _authRequestType == AuthRequestType.AdminApproval)
             {
-                //TODO TDE if is admin type save to memory to later see if it was approved
-                /*
-                  const adminAuthReqStorable = new AdminAuthRequestStorable({
-                      id: reqResponse.id,
-                      privateKey: this.authRequestKeyPair.privateKey,
-                    });
-
-                    await this.stateService.setAdminAuthRequest(adminAuthReqStorable);
-                */
-                FingerprintPhrase = response.FingerprintPhrase;
-                _requestId = response.Id;
-                _requestAccessCode = response.RequestAccessCode;
-                _requestKeyPair = response.RequestKeyPair;
+                response = await _authService.GetPasswordlessLoginRequestByIdAsync(pendingRequest.Id);
+                if (response == null || (response.IsAnswered && !response.RequestApproved.Value))
+                {
+                    // handle pending auth request not valid remove it from state
+                    await _stateService.SetPendingAdminAuthRequestAsync(null);
+                    pendingRequest = null;
+                }
+                else
+                {
+                    // Derive pubKey from privKey in state to avoid MITM attacks
+                    // Also generate FingerprintPhrase locally for the same reason
+                    var derivedPublicKey = await _cryptoFunctionService.RsaExtractPublicKeyAsync(pendingRequest.PrivateKey);
+                    response.FingerprintPhrase = string.Join("-", await this._cryptoService.GetFingerprintAsync(Email, derivedPublicKey));
+                    response.RequestKeyPair = new Tuple<byte[], byte[]>(derivedPublicKey, pendingRequest.PrivateKey);
+                }
             }
 
+            if (response == null)
+            {
+                response = await _authService.PasswordlessCreateLoginRequestAsync(_email, AuthRequestType);
+            }
+
+            await HandlePasswordlessLogin(response, pendingRequest == null && _authRequestType == AuthRequestType.AdminApproval);
             await _deviceActionService.HideLoadingAsync();
         }
 
+        private async Task HandlePasswordlessLogin(PasswordlessLoginResponse response, bool createPendingAdminRequest )
+        {
+            if (response == null)
+            {
+                throw new Exception(AppResources.GenericErrorMessage);
```

Comment:
⛏️ I'd throw an `ArgumentNullException` here, this is useful to us on the error report and `AppResources.GenericErrorMessage` doesn't provide value to us for analysis, additionally this message won't be seen by the user, given the `HandleException` logic.
```suggestion
                throw new ArgumentNullException(nameof(response));
```